app/controllers/api/v2/hosts_controller.rb

```diff
@@ -353,6 +341,13 @@ def templates
         end
       end
 
+      api :GET, "/hosts/:id/virtualization", N_("Get virtualization information for the host")
```

Comment:
I don't see the benefit of this over expanding the `vm_compute_attributes. Could you explain why this needs a new endpoint? If you can't and given that this includes the VM attributes, should we deprecate the old endpoint?

Reply:
This new endpoint needs both `vm` and `provider` for the virtualization card. There is no way to find out the vendor type from `vm` itself. 
Not sure who are using the old endpoint which may need some investigation. And its deletion is beyond the scope of this PR.

Reply:
> This new endpoint needs both vm and provider for the virtualization card. There is no way to find out the vendor type from vm itself.

I don't see any consideration about adding the provider to the existing endpoint. Has that been considered as an alternative?

>  Not sure who are using the old endpoint which may need some investigation. And its deletion is beyond the scope of this PR.

The actual removal may be beyond the scope, but IMHO it must be discussed. We can't just keep adding new API entry points all the time without careful design.

Reply:
> I don't see any consideration about adding the provider to the existing endpoint. Has that been considered as an alternative?

This would change the format of the return value. Currently it returns the VM info as a key-value hash. 
Adding provider would make the return value as `{vm: vm, provider: provider}`.  Not sure if this change would break the code for existing users.

If the format change is not a concern, I would prefer expand this endpoint.

> The actual removal may be beyond the scope, but IMHO it must be discussed. We can't just keep adding new API entry points all the time without careful design.

Totally agree. Thanks for the review.

Reply:
> If the format change is not a concern, I would prefer expand this endpoint.

No, you're right that a format change is a bad thing in the API. But I do wonder if the provider could be considered an attribute by itself.

Reply:
How about I insert it as `vendor_type: @host.compute_resource.provider.downcase` into the VM attributes in old endpoint? In this way, no need to add a new endpoint.

Reply:
I'll be honest and say I don't exactly know which attributes we currently return, but that is indeed what I was thinking about. The alternative is to add properties to the host. We already have `compute_resource_id` and `compute_resource_name` there (according to https://apidocs.theforeman.org/foreman/latest/apidoc/v2/hosts/show.html) so you could also add `compute_resource_provider`. Since it has to provide the name, you probably don't perform an additional DB query so the cost is small.

Reply:
I like the idea of `compute_resource_provider`. Will go with it.

Reply:
As also said privately on IRC: you can also consider a `compute_resource` node that has the ID, name and provider. Possibly even (conditionally) include the whole compute resource. However, do consider the cost. Try to avoid any additional queries (either DB or to an external resource).